kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -321,7 +321,38 @@ impl<'tcx> GotocCtx<'tcx> {
     ///
     /// Handle all attributes i.e. `#[kani::x]` (which kani_macros translates to `#[kanitool::x]` for us to handle here)
     fn handle_kanitool_attributes(&mut self) {
-        let all_attributes = self.tcx.get_attrs_unchecked(self.current_fn().instance().def_id());
+        let def_id = self.current_fn().instance().def_id();
+        let all_attributes = self.tcx.get_attrs_unchecked(def_id);
+
+        if def_id.is_local() {
```

Comment:
This code seems out of place. You are looking for things specific to `#[test]` attributes inside a function that looks for `kanitool` attributes.

Can you also move this code to its own function?